### PR TITLE
Adding back save api

### PR DESCRIPTION
**Pull request by rajmusuku**
Adding back save method to following files 
azdata.proposed.d.ts
sqlops.proposed.d.ts
notebook.component.ts

### CODE REVIEW

File: src/sql/parts/notebook/notebook.component.ts [hunk 1]
```diff
@@ -526,6 +424,15 @@ export class NotebookComponent extends AngularDisposable implements OnInit, OnDe
 		provider.bindTo(this.contextKeyService);
 	}
 
+	public async save(): Promise<boolean> {
+		try {
+			return this._notebookParams.input.save();
+		} catch (err) {
+			this.notificationService.error(localize('saveFailed', 'Failed to save notebook: {0}', notebookUtils.getErrorMessage(err)));
+			return false;
```
Comment (kevcunnane) [inline src/sql/parts/notebook/notebook.component.ts]: should this be Promise.resolve(false)? Or else don't try / catch and just return to caller who can handle the rejection in whatever way they want?

File: src/sql/parts/notebook/notebook.component.ts [hunk 2]
```diff
@@ -526,6 +424,15 @@ export class NotebookComponent extends AngularDisposable implements OnInit, OnDe
 		provider.bindTo(this.contextKeyService);
 	}
 
+	public async save(): Promise<boolean> {
+		try {
+			return this._notebookParams.input.save();
+		} catch (err) {
```
Comment (kevcunnane) [inline src/sql/parts/notebook/notebook.component.ts]: I'd suggest don't handle rejection, as the core save implementation does most of this. The caller should expect potential for rejected promise and be able to decide what to do itself.

File: src/sql/workbench/api/node/sqlExtHost.protocol.ts [hunk 3]
```diff
@@ -860,7 +860,6 @@ export interface ExtHostNotebookDocumentsAndEditorsShape {
 }
 
 export interface MainThreadNotebookDocumentsAndEditorsShape extends IDisposable {
-	$trySaveDocument(uri: UriComponents): Thenable<boolean>;
```
Comment (kevcunnane) [inline src/sql/workbench/api/node/sqlExtHost.protocol.ts]: None of these should be gone - how is stuff compiling?

File: src/sql/parts/notebook/notebook.component.ts (hunk 1 above)
Comment (rajmusuku) [reply src/sql/parts/notebook/notebook.component.ts]: removed exception handling

File: src/sql/workbench/api/node/sqlExtHost.protocol.ts (hunk 3 above)
Comment (rajmusuku) [reply src/sql/workbench/api/node/sqlExtHost.protocol.ts]: Good to know that I have to kill existing process and rebuild it again. Made changes.

File: src/sql/parts/notebook/notebook.component.ts [hunk 4]
```diff
@@ -526,6 +424,10 @@ export class NotebookComponent extends AngularDisposable implements OnInit, OnDe
 		provider.bindTo(this.contextKeyService);
 	}
 
+	public async save(): Promise<boolean> {
```
Comment (kevcunnane) [inline src/sql/parts/notebook/notebook.component.ts]: Seems like you don't need this as you implemented it in MainThreadNotebookDocumentsAndEditors with a direct call through to the input. I'd suggest removing.

**Pull request merged**